staging/src/k8s.io/apiserver/pkg/endpoints/handlers/fieldmanager/scalehandler.go

```diff
@@ -68,7 +68,12 @@ func (h *ScaleHandler) ToSubresource() ([]metav1.ManagedFieldsEntry, error) {
 	f := fieldpath.ManagedFields{}
 	t := map[string]*metav1.Time{}
 	for manager, versionedSet := range managed.Fields() {
-		path := h.mappings[string(versionedSet.APIVersion())]
+		path, ok := h.mappings[string(versionedSet.APIVersion())]
+		// Drop the field if the APIVersion of the managed field is unknown
+		if !ok {
```

Comment:
My concern with this is that we will remove entries for mappings that don't exist, not really for versions that don't exist. What if forgot to create a mapping? How would we know other than because this triggers a bug of managedFields being removed?

Reply:
This is totally a valid point, I am not sure how to check that though 🤔

Reply:
OK I've created a test for `ReplicaSet`:
```golang
func TestReplicasForAllVersions(t *testing.T) {
	knownVersions := scheme.Scheme.VersionsForGroupKind(schema.GroupKind{Group: "apps", Kind: "ReplicaSet"})
	for _, version := range knownVersions {
		if _, ok := replicasPathInReplicaSet[version.String()]; !ok {
			t.Fatalf("missing version %v in `replicasPathInReplicasSet`", version)
		}
	}

	if len(knownVersions) != len(replicasPathInReplicaSet) {
		t.Fatalf("`replicasPathInReplicaSet` has extra items: %v vs %v", replicasPathInReplicaSet, knownVersions)
	}
}
```

It's failing because we don't have apps/v1beta Replicaset, looks like this was useful :-)

Reply:
I used `k8s.io/client-go/kubernetes/scheme` for the scheme by the way.

Reply:
Thanks a lot @apelisse ! 🎉  I have add the tests in https://github.com/kubernetes/kubernetes/pull/98377/commits/cc220b6ea4cdcff66d415be2a925bb88367b76c1